gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -49,28 +51,95 @@ public void write(Bundle bundle, File rootFolder) {
         });
 
         //create policies
+        Map<String, PolicyMetadata> policyMetadataMap = new HashMap<>();
         Map<String, Service> services = bundle.getEntities(Service.class);
-        services.values().parallelStream().forEach(serviceEntity -> writePolicy(bundle, policyFolder, serviceEntity.getParentFolder().getId(), serviceEntity.getName(), serviceEntity.getPolicyXML()));
+        services.values().parallelStream().forEach(serviceEntity -> {
```

Comment:
I would like not to pollute **writePolicy** method with PolicyMetadata creation. I don't see many dependencies in doing so. 
I would do it like below:

``
writePolicy(...
map.put(path+name, createPolicyMetadata(id, name, bundle))
``

Reply:
done